x/nexus/keeper/general_message.go

```diff
@@ -232,6 +234,31 @@ func (k Keeper) validateAddressAndAsset(ctx sdk.Context, address exported.CrossC
 	return k.validateAsset(ctx, address.Chain, asset.Denom)
 }
 
+func (k Keeper) getRouteMessageQueue(ctx sdk.Context) utils.KVQueue {
+	return utils.NewBlockHeightKVQueue(routeMessageQueue, k.getStore(ctx), ctx.BlockHeight(), k.Logger(ctx))
+}
+
+// EnqueueRouteMessage enqueues the given general message to be routed
+func (k Keeper) EnqueueRouteMessage(ctx sdk.Context, id string) error {
+	msg, ok := k.GetMessage(ctx, id)
+	if !ok {
+		return fmt.Errorf("general message %s not found", id)
+	}
+
+	if !(msg.Is(exported.Approved) || msg.Is(exported.Failed)) {
+		return fmt.Errorf("general message has to be approved or failed")
+	}
```

Comment:
why can failed messages get routed?

Reply:
Failed messages can always be routed for re-try. This is the same behaviour as [RouteMessage](https://github.com/axelarnetwork/axelar-core/blob/f6bc799eebafd453082e7431ec70112fe24ce89d/x/nexus/keeper/general_message.go#L264).